components/autocomplete_selector.tsx

```diff
@@ -32,47 +49,41 @@ export default class AutocompleteSelector extends React.PureComponent {
         listComponent: SuggestionList,
         listPosition: 'top',
     };
+    suggestionRef?: React.RefObject<HTMLDivElement>;
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
         this.state = {
             input: '',
         };
     }
 
-    onChange = (e) => {
+    onChange = (e: ChangeEvent<HTMLInputElement>) => {
         if (!e || !e.target) {
             return;
         }
 
         this.setState({input: e.target.value});
     }
 
-    handleSelected = (selected) => {
+    handleSelected = (selected: Selected) => {
         this.setState({input: ''});
 
-        if (this.props.onSelected) {
-            this.props.onSelected(selected);
-        }
+        this.props.onSelected?.(selected);
```

Comment:
actually we dont need to change these. But its fine.

Reply:
Yes, these changes based on my feedback. Since we are using TS, it would be good to use optional chaining to reduce number of lines. But you are right, old and new versions are basically same.

Reply:
yeah no worries